src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs

```diff
@@ -105,6 +105,11 @@ private void OnSettingsPropertyChanged(object sender, System.ComponentModel.Prop
                 SDK.AllowLocalAdaptersWhenConnected = Properties.Settings.Default.AllowLanInterfaces;
                 Properties.Settings.Default.Save();
             }
+
+            if (e.PropertyName == nameof(Properties.Settings.Default.IsThreatProtectionEnabled))
+            {
+                SDK.DnsFilterMode = IsThreatProtectionEnabled ? VpnSDK.Enums.DnsFilteringMode.WithPartnerDns : VpnSDK.Enums.DnsFilteringMode.Disabled;                 
```

Comment:
@renishnair-viper this needs to be 

SDK.DnsFilterMode = IsThreatProtectionEnabled ? VpnSDK.Enums.DnsFilteringMode.WithWLVPNDns: VpnSDK.Enums.DnsFilteringMode.Disabled; so the code is in sync with https://github.com/wlvpn/ConsumerVPN-Windows/pull/14/files#diff-0633ab59162df35a92a0f89f6f2604aa97ad69fddf1359704cd3c1d0d1d33560R104

Reply:
Updated SDK.DnsFilterMode as per suggestion